app/assets/javascripts/views/questioning_form_view.js.coffee

```diff
@@ -30,7 +30,7 @@ class ELMO.Views.QuestioningFormView extends ELMO.Views.QuestionFormView
     @showDefault() && (@fieldValue('default') || '').trim() != ''
 
   showRequired: ->
-    !@fieldValue('read_only') && @super.metadataTypeBlank.call(this)
+    (!@fieldValue('read_only') || @fieldValue('read_only') == 'No') && @super.metadataTypeBlank.call(this)
```

Comment:
I couldn't determine why this regressed (or maybe it's always been this way) but this seems safe. More risky would be to remove the first part of the condition if we know they'll all be `No` (rather than empty or `false`), but I didn't want to make that assumption.

Reply:
I figured out what is going on here. See `app/helpers/elmo_form_builder.rb:178`. questionings.read_only can either be nil or false in the negative case. There is no null constraint on the DB b/c read_only doesn't apply to QingGroups. This may have been a mistake. But I don't think it's worth trying to change that right now. May cause weird side effects. 

If it's false, things work as planned. If it's `nil`, then the `data-val` attrib doesn't get included and that breaks the `@fieldValue` check. The solution I think is to change `val` from `nil` to `false` around line 149 of that file. Let me know what you think...

Reply:
Wonderful, simpler fix implemented (and fixup'd into commit history). Thanks for diagnosing this.